### PR TITLE
Update notes.xsl

### PR DESCRIPTION
**Pull request by AurelioDeRosa**
Fix issue #586.

### CODE REVIEW

**Comment by arthurvr** [comment]
The `data-doesnt-accept-undefined` note does [also appear](https://github.com/jquery/api.jquery.com/blob/bb5a21d6215d229190182b52bd77ffc98d046e23/entries/jQuery.data.xml#L28) on the `jQuery.data` post. Of course, `jQuery.data` [doesn't return the jQuery object](http://jsbin.com/goyajiwuhu/1/edit?js) is is called on.

**Comment by AurelioDeRosa** [comment]
@arthurvr In this case I vote for splitting the two notes as they have a different behavior.

**Comment by arthurvr** [comment]
Seems fine.

**Comment by dmethvin** [comment]
Seems like we need to split the two, I agree. How do you guys feel about `undefined` not actually setting the value? 

> recognised

Have we standardised on UK English? Oh I mean, *standardized*. :smile: I guess I never noticed before.

**Comment by AurelioDeRosa** [comment]
To be honest I feel more like `.data('thing', undefined)` should act as `.removeData('thing')`. We had a similar discussion in #523.

P.S. British English vs American English: their differences will kill me.

**Comment by dmethvin** [comment]
I had forgotten about #523. It's tough because I think a good argument can be made either way. I agree with you that removing the data item feels best, but it would certainly be a breaking change as @gibson042 pointed out. It's hard to know whether a lot of plugins and developers depend on the current no-op behavior.

**Comment by AurelioDeRosa** [comment]
@dmethvin What if we introduce this BC in jQuery 3.0?

**Comment by dmethvin** [comment]
If very few people depend on this (or those people don't upgrade) then it would be nice to eliminate this kind of strange exception. If it turns out that several high-profile popular plugins break because we change this, then it would be a real pain to do and it would prevent people from upgrading to 3.0. I don't know which applies. :smile:

**Comment by AurelioDeRosa** [comment]
@dmethvin we're all in your hands :)

**Comment by dmethvin** [comment]
I'll leave the hard decisions to @timmywil now! Sucker!

**Comment by arthurvr** [comment]
@AurelioDeRosa Besides from the discussion above, this PR needs some updates. Would you like to do them? I'd love to land this.

**Comment by AurelioDeRosa** [comment]
@arthurvr I was waiting for a feedback by @timmywil.

**Comment by timmywil** [comment]
Sorry I missed this. I think this can be landed. As far as #523, I think we'll need to gauge how much it will break. Personally, I always saw undefined as it's own value and thought it was most intuitive for `.data('something', undefined)` to act similarly to assigning a property on an object to the value `undefined`, and `.removeData('something')` being the equivalent of using the `delete` operator. Otherwise, determining whether something was set to `undefined` or whether it was never set is impossible. Regardless, we can thrash on that issue there.

**Comment by arthurvr** [comment]
I can def. agree with @timmywil.

**Comment by AurelioDeRosa** [comment]
So, I understand the conclusion is "Let's leave things as they are in regard of `undefined` and split the note in two notes". I'll update my PR accordingly.

**Comment by arthurvr** [comment]
Thank you, @AurelioDeRosa!

**Comment by arthurvr** [comment]
> I'll update my PR accordingly.

@AurelioDeRosa You've got the chance to update this PR? Would love to land it.

**Comment by arthurvr** [comment]
ping @AurelioDeRosa

**Comment by AurelioDeRosa** [comment]
PR updated @arthurvr

File: notes.xsl [hunk 1]
```diff
@@ -2,6 +2,9 @@
 <xsl:template name="note">
 	<xsl:choose>
 		<xsl:when test="@id = 'data-doesnt-accept-undefined'">
+			<code>undefined</code> is not recognised as a data value. Calls such as <code><xsl:value-of select="@data-title" />( <xsl:value-of select="@data-parameters" />, undefined )</code> will return the jQuery object that it was called on, allowing for chaining.
```
Comment (arthurvr) [inline notes.xsl]: > recognised

We use American English. Could you fix that on the while in the `jquery-data-doesnt-accept-undefined` note.

**Pull request closed without merging**